source/blog/2015-05-08-rspec-compsable-matchers.md

```diff
@@ -0,0 +1,216 @@
+---
+title: How RSpec 3.0 composable matchers work
+author: Sam Phippen
+---
+
+##Warmup: the `===` protocol
+
+In Ruby, a number of objects implement the `===` protocol. Some of the objects
+that implement this protocol include:
+
+* all class objects
+* lambdas
+* regexes
+* ranges
+
+Conceptually, this protocol represents a "match". That is, the object that
+implements the `===` declares whether or not it "matches" the other object.
+For example, `Array === []` returns `true`. This is because an empty array is a
+subclass of the array class. `Array === {}` returns false because a hash is not
```

Comment:
Shouldn't this be:

> This is because an empty array is an instance of the `Array` class.

...rather than:

> This is because an empty array is a subclass of the array class.

Given that every object has a singleton class (which is a one-off subclass of the instantiated class) the way you've phrased it isn't entirely inaccurate, but it's not how ruby programmers usually explain it.